### PR TITLE
feat: add offline-ras synthesize output format

### PR DESCRIPTION
**Pull request by joanise**
### PR Goal? 

Add the ability to synthesize directly to readalong's Offline HTML format.

### Fixes? 

Enhancement to #102 to complete fixing https://github.com/EveryVoiceTTS/EveryVoice/issues/439

### Feedback sought? 

test, and review code

### Priority? 

beta

### Tests added? 

yes

### How to test? 

    everyvoice synthesize from-text  -O offline-ras -t "stuff and nonsense" -v <vocoder.cpkt> <fs2.ckpt>

and look at the synthesized readalong/*.html file.

"forget" the -v option and see a nice error message

### Confidence? 

pretty good

### Version change? 

it'll contribute to the minor we're working on.

### Related PRs? 

Coming soon to a PR near you...
https://github.com/EveryVoiceTTS/EveryVoice/pull/608
https://github.com/ReadAlongs/Studio/pull/256

### CODE REVIEW

File: fs2/cli/synthesize.py [hunk 1]
```diff
@@ -342,10 +342,21 @@ def synthesize(  # noqa: C901
...
+        '**spec**' will generate predicted Mel spectrograms. Tensors are Mel band-oriented (K, T) where K is equal to the number of Mel bands and T is equal to the number of frames.
+
+
         '**textgrid**' will generate a Praat TextGrid with alignment labels. This can be helpful for evaluation.
+
+
         '**readalong**' will generate a ReadAlong from the given text and synthesized audio (see https://github.com/ReadAlongs).
+
+
+        '**offline-ras**' will generate a single file Offline HTML ReadAlong that can be further edited in the ReadAlong Studio Editor. Also implies '--output-type wav'. Requires --vocoder-path.
```
Comment (roedoejet) [inline fs2/cli/synthesize.py:359]: I would prefer to call this `readalong-html` just for continuity with the other one. maybe calling the other one `readalong-xml` or something?

File: fs2/prediction_writing_callback.py [hunk 2]
```diff
@@ -350,7 +369,15 @@ def __init__(
         self.output_key = output_key
         logger.info(f"Saving pytorch output to {self.save_dir}")
 
-    def save_aligned_text_to_file(self, max_seconds, phones, words, language, filename):
+    def save_aligned_text_to_file(
+        self,
+        max_seconds: float,
```
Comment (roedoejet) [inline fs2/prediction_writing_callback.py:374]: nice improvement to the fn signature

File: fs2/prediction_writing_callback.py [hunk 3]
```diff
@@ -309,7 +319,15 @@ def __init__(
             save_dir=output_dir / "textgrids",
         )
 
-    def save_aligned_text_to_file(self, max_seconds, phones, words, language, filename):
+    def save_aligned_text_to_file(
+        self,
+        max_seconds: float,
```
Comment (roedoejet) [inline fs2/prediction_writing_callback.py:324]: nice improvement

**Review by roedoejet — CHANGES_REQUESTED** [review]
Looks good to me, @joanise - thanks! I would request changing the `offline-ras` variable name to something with more continuity with the existing `readalong` output type. Either with both `readalong-xml` and `readalong-html` or just `readalong` for the html/offline and `readalong-xml`.

File: fs2/cli/synthesize.py (hunk 1 above)
Comment (joanise) [reply fs2/cli/synthesize.py:359]: Yeah, makes sense. I think I'll go for `readalong-html` and `readalong-xml`.

Comment (joanise) [reply fs2/cli/synthesize.py]: Done

**Review by roedoejet — APPROVED** [review]
awesome thanks!

**Pull request merged**